### PR TITLE
bugfix: fix the exception in RedisTransactionStoreManager

### PR DESCRIPTION
**Pull request by wingchi-leung**
- [x] I have registered the PR [changes](https://github.com/seata/seata/tree/develop/changes).

### Ⅰ. Describe what this PR did
修复RedisTransactionStoreManager中遍历Map时修改，抛出ConcurrentModificationException的问题。

### Ⅱ. Does this pull request fix one issue?

fixes: https://github.com/seata/seata/issues/4667

### CODE REVIEW

File: server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java [hunk 1]
```diff
@@ -750,7 +753,7 @@ private void dogetXidsForTargetMapRecursive(Map<String, Integer> targetMap, long
                     listList.add(list);
```
Comment (slievrly) [inline server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java:753]: `sum < queryCount` always true.

File: server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java [hunk 2]
```diff
@@ -750,7 +753,7 @@ private void dogetXidsForTargetMapRecursive(Map<String, Integer> targetMap, long
...
                 } else {
                     if (list.size() == 0) {
-                        targetMap.remove(key);
+                        iterator.remove();
```
Comment (slievrly) [inline server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java:756]: this looks good.

**Comment by slievrly** [comment]
@wingchi-leung Pls add changes.
- https://github.com/seata/seata/blob/develop/changes/en-us/develop.md
- https://github.com/seata/seata/blob/develop/changes/zh-cn/develop.md

File: server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java (hunk 1 above)
Comment (wingchi-leung) [reply server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java:753]: ok

**Review by slievrly — APPROVED** [review]
LGTM

**Review by a364176773 — COMMENTED** [review]
/home/runner/work/seata/seata/server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java:737:13: 'while' is not followed by whitespace. [WhitespaceAround]

**Review by a364176773 — APPROVED** [review]
LGTM

**Pull request merged**